### ISSUE TITLE
[🐛] Build error from flag-icons 6.9.x dependency: Can't resolve flag path for flags-icons.scss

### ISSUE THREAD

**Issue by unixerius** [juice-shop/juice-shop#2047]
# :bug: Bug report

## Description

I am running Ubuntu 22.04 with the most recent Docker available to me. 

I have cloned a fresh copy of the Juice-Shop repository and am running `docker build -t juiceshop:myteam .`

The Docker build starts but crashes during step 5, which does all the NPM builds. 

While building the frontend, the build dies with an exitcode 1 and the following errors:

<pre><code>
./node_modules/flag-icons/sass/flag-icons.scss - Error: Module Error (from ./node_modules/postcss-loader/dist/cjs.js):
/juice-shop/frontend/node_modules/flag-icons/sass/flag-icons.scss:1800:25: Can't resolve '../flags/4x3/ea.svg' in '/juice-shop/frontend/node_modules/flag-icons/sass'

./node_modules/flag-icons/sass/flag-icons.scss - Error: Module Error (from ./node_modules/postcss-loader/dist/cjs.js):
/juice-shop/frontend/node_modules/flag-icons/sass/flag-icons.scss:1803:25: Can't resolve '../flags/1x1/ea.svg' in '/juice-shop/frontend/node_modules/flag-icons/sass'
</code></pre>

### Is this a regression?

Yes. I do not remember which version of Juiceshop it was, but when I last run this experiment in Q1 2023, the build worked fine. So it worked at least four months ago. 

I have also tested with v14.5.1. The issue also exists there. I will slowly start making my way backwards to when it did work.

**EDIT:** Also happens for me with 14.4.0.

**EDIT:** I must be doing something very wrong, because I could have sworn it used to work with 14.3.0. But now I'm getting the same errors over there. 

## :microscope: Minimal Reproduction

1. Use Vagrant to build an Ubuntu 22.04 host. 
2. Vagrantfile can be found here -> https://github.com/unixerius/DSO/blob/main/Lab%20setup/Vagrantfile
3. Git clone the juiceshop repo.
4. Run `docker build -t juiceshop:myteam .` inside the juiceshop repo.

## :fire: Exception or Error

<pre><code>
./node_modules/flag-icons/sass/flag-icons.scss - Error: Module Error (from ./node_modules/postcss-loader/dist/cjs.js):
/juice-shop/frontend/node_modules/flag-icons/sass/flag-icons.scss:1800:25: Can't resolve '../flags/4x3/ea.svg' in '/juice-shop/frontend/node_modules/flag-icons/sass'

./node_modules/flag-icons/sass/flag-icons.scss - Error: Module Error (from ./node_modules/postcss-loader/dist/cjs.js):
/juice-shop/frontend/node_modules/flag-icons/sass/flag-icons.scss:1803:25: Can't resolve '../flags/1x1/ea.svg' in '/juice-shop/frontend/node_modules/flag-icons/sass'</code></pre>

## :deciduous_tree: Your Environment

The command suggested (`node -v && npm -v`) does not apply to the VM in question. So I ran it inside the build container.

<pre><code>
vagrant@devsecops:~/juicy$ docker run node:18 node -v
v18.17.0

vagrant@devsecops:~/juicy$ docker run node:18 npm -v
9.6.7
</code></pre>

### Additional Information

I have found that the build works, if I first run `npm install` in the build context. 

This apparently fetches a few extra dependencies which would otherwise not make it into the build context, thus leading to the faulty situation. 

Now it's time to find out which deps are missing / broken when we _don't_ first run `npm install` outside the container.

**Comment by unixerius** [comment]
I've done the same test with 15.0.0, 14.5.1, 14.4.0 and 14.3.0. All refuse to `docker build` properly, unless I first `npm install` in the Git repository. Apparently there's some dependency that's not getting done properly. I just don't know what, because the build used to work fine early this year.

**Comment by unixerius** [comment]
Editing the Dockerfile, changing line 5 from `RUN npm install --omit=dev --unsafe-perm` into `RUN npm install` didn't help anything. Still fails on those two flags, but since I'm really not a developer I have no clue what's happening.

**Comment by unixerius** [comment]
What in tarnation? Now I've accidentally reproduced the same build error outside of Docker and inside my VM. This is bonkers. 

Attaching the build log here. 

[2023-07-25T19_41_21_825Z-debug-0.log](https://github.com/juice-shop/juice-shop/files/12165134/2023-07-25T19_41_21_825Z-debug-0.log)

Plus another one. Now it even reliably happens on my Ubuntu VM, not just inside Docker. 

[2023-07-25T19_54_25_176Z-debug-0.log](https://github.com/juice-shop/juice-shop/files/12165263/2023-07-25T19_54_25_176Z-debug-0.log)

**Comment by bkimminich** [comment]
This seems a problem with the `flag-icons` library that occured in `6.9.0` and was still present in `6.9.1`. Our dependency pulled the latest minor `6.x` version but I changed it on `develop` branch now to pull `6.8.x` versions only. With that, installation works, and therefore also Docker image builds should be fine again. I won't do a bugfix release for this, as the Flag Icons folks might be on top of this soon anyway.

**TL;DR:** Please use our `develop` branch to test your Docker build and let me know if it works now. Cheers!

**Comment by bkimminich** [comment]
See also https://github.com/lipis/flag-icons/issues/1120

**Comment by unixerius** [comment]
Aha! I was thinking it was something like that, but I didn't know how to change it myself. I will defo give this a quick test, thank you! I'll update asap.

**Comment by unixerius** [comment]
1. Thank you very much for your very swift help. I really appreciate it, you've saved my bacon for tomorrow's class. :)
2. It seems that the build is getting further along now. There's plenty of warnings about  `error TS7016: Could not find a declaration file for module 'express'` for many of the routes. But they don't seem to block the build. 

Building on Azure DevOps takes about 12 minutes... That's a lot more than it used to; builds used to sit around 7-8 minutes. But, I'll take it.
